### ISSUE TITLE
Issue chunked request cause warning "Connection pool is full"

### ISSUE THREAD

**Issue by ddzialak** [kennethreitz/requests#3531]
```
import requests
import sys
import logging
logging.basicConfig(stream=sys.stdout, level=logging.DEBUG)
logger=logging.getLogger()

def produce():
    for _ in xrange(5000):
        yield "bla" * 1000 + '\0\n'


logger.info("sending....")
response = requests.request("POST", "http://localhost:30006/upload", data=produce(), stream=True)
logger.info("received respone: %s", response.json())
```

As the result I'm receiving:
```
python test.py 
INFO:root:sending....
INFO:requests.packages.urllib3.connectionpool:Starting new HTTP connection (1): localhost
**WARNING:requests.packages.urllib3.connectionpool:Connection pool is full, discarding connection: localhost**
INFO:root:received respone: {u'estimation': 0, u'reason_code': None,
```

Everything works but there should be no warning.

**Comment by ddzialak** [comment]
That happens because `_put_conn` (from connectionpool.py line 259) is called twice, one from:

```
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/api.py", line 44, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/sessions.py", line 456, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/sessions.py", line 559, in send
    r = adapter.send(request, **kwargs)
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/adapters.py", line 369, in send
    conn._put_conn(low_conn)
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/packages/urllib3/connectionpool.py", line 259, in _put_conn
```

 AND second time from:

```
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/api.py", line 44, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/sessions.py", line 456, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/sessions.py", line 596, in send
    r.content
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/models.py", line 694, in content
    self._content = bytes().join(self.iter_content(CONTENT_CHUNK_SIZE)) or bytes()
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/models.py", line 627, in generate
    for chunk in self.raw.stream(chunk_size, decode_content=True):
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/packages/urllib3/response.py", line 240, in stream
    data = self.read(amt=amt, decode_content=decode_content)
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/packages/urllib3/response.py", line 221, in read
    self.release_conn()
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/packages/urllib3/response.py", line 125, in release_conn
    self._pool._put_conn(self._connection)
  File "/home/dzialak/.virtualenvs/sf3/lib/python3.5/site-packages/requests/packages/urllib3/connectionpool.py", line 267, in _put_conn
```

**Comment by ddzialak** [comment]
It's about requests (2.3.0)

**Comment by ddzialak** [comment]
Closing that issue as it's already solved in 2.11.1
